Automattic/woocommerce-payments
# Add all subscription line items when creating WCPay Subscriptions

Fixes #3826 

#### Changes proposed in this Pull Request

While testing compatibility with 3rd party plugins such as [**All Products for Subscriptions**](https://woocommerce.com/products/all-products-for-woocommerce-subscriptions/) and [**Composite Products**](https://woocommerce.com/products/composite-products/) I discovered a bug when creating WCPay Subscriptions [we only send line item data if the line item product is a subscription product type.](https://github.com/Automattic/woocommerce-payments/blob/3.7.0/includes/subscriptions/class-wc-payments-subscription-service.php#L735-L737) 

This causes some edge case issues where a subscription in WC doesn't contain just subscription products, as is the case with manually created subscriptions or complex scenarios like composite or bundled products. 

This PR fixes that by making sure we send line item data for all subscription line items even if they aren't a subscription product. This is based on the principle that if it's a line item for a subscription, it's recurring in this context even if it's not a subscription product.

#### Testing instructions

* With WC Pay Subscriptions (the feature enabled), create a simple and at least 1 subscription product. 
* Go to **WooCommerce > Subscriptions > Create new**.
* Select your current user as the customer.
* Add the simple product you created in step 1 as a line item
* Add the subscription product as well.
* Set the status to **Active**. 
* Go to your Subscriptions section on your Account page 
* Find the subscription you just created and attempt to add a payment method too it. 
* Note that when you land on the change payment screen the items. 
* Select a card and click to add the payment method.
* View the subscription in Stripe and notice that only the subscription product got added.
    - With this branch, all subscription product get added to the subscription.

<img width="1097" alt="Screen Shot 2022-02-25 at 2 31 24 pm" src="https://user-images.githubusercontent.com/8490476/155655020-efaccd59-ecbb-401c-8b0b-02955e497f83.png">

---

<img width="1428" alt="Screen Shot 2022-02-25 at 2 38 59 pm" src="https://user-images.githubusercontent.com/8490476/155655009-2cefe485-7b05-43f3-99c9-deb5027e927f.png">

_Notice that only the weekly product got added to the subscription._

-------------------


**Post merge**
